cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -174,10 +174,10 @@ impl Inst {
         Inst::Div {
             size,
             signed,
-            divisor,
-            dividend: regs::rax(),
-            dst_quotient: Writable::from_reg(regs::rax()),
-            dst_remainder: Writable::from_reg(regs::rdx()),
+            divisor: GprMem::new(divisor).unwrap(),
+            dividend: Gpr::new(regs::rax()).unwrap(),
```

Comment:
Would it make sense to make `regs::rax()` and friends return a `Gpr` directly, and likewise `regs::xmm0()` and friends return an `Xmm` directly? This callsite gets simpler but others maybe gain a `.to_reg()`, so it's not necessarily any less verbose, but it is a bit more typesafe.

Reply:
I've avoided pushing the newtypes beyond the ISLE-glue boundary thus far. I think it would make sense to have one more PR (maybe I'll get to it while waiting on reviews for my components PRs) where we push the newtypes to all the `Inst::add`/etc methods and I think pushing that to `regs::rax`/etc in that same PR would make sense.

But I kinda don't want to do all that right now.

Does that sound good?

Reply:
Yep, that's reasonable, any incremental increase in typed-ness is good but no need to block on any part of it!